x/auth/legacy/v043/store.go

```diff
@@ -0,0 +1,266 @@
+package v043
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/vesting/exported"
+	vestingtypes "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/gogo/protobuf/grpc"
+	"github.com/gogo/protobuf/proto"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	delegatorDelegationPath           = "/cosmos.staking.v1beta1.Query/DelegatorDelegations"
+	stakingParamsPath                 = "/cosmos.staking.v1beta1.Query/Params"
+	delegatorUnbondingDelegationsPath = "/cosmos.staking.v1beta1.Query/DelegatorUnbondingDelegations"
+	balancesPath                      = "/cosmos.bank.v1beta1.Query/AllBalances"
```

Comment:
As noted by @aaronc, another way to remove these hardcoded paths is to use `QueryServiceTestHelper`.

So MigrateStore takes an interfaceRegistry instead:
```diff
- func MigrateStore(ctx sdk.Context, accounts []types.AccountI, queryServer grpc.Server) ([]types.AccountI, error) {
+ func MigrateStore(ctx sdk.Context, accounts []types.AccountI, interfaceRegistry codectypes.InterfaceRegistry) ([]types.AccountI, error) {
-	return migrateVestingAccounts(ctx, accounts, queryServer)
+	return migrateVestingAccounts(ctx, accounts, interfaceRegistry)}
```

We then create a grpc client connection for the purpose of the migration:
```go
func migrateVestingAccounts(ctx sdk.Context, accounts []types.AccountI, interfaceRegistry codectypes.InterfaceRegistry) ([]types.AccountI, error) {
	grpcClientConn := baseapp.NewQueryServerTestHelper(ctx, interfaceRegistry)

	bondDenom, err := getBondDenom(ctx, grpcClientConn)
```

and getBondDenom would look much nicer:

```go
func getBondDenom(ctx sdk.Context, clientConn grpc.ClientConn) (string, error) {
	stakingClient := stakingtypes.NewQueryClient(clientConn)

	res, err := stakingClient.Params(context.Background(), &stakingtypes.QueryParamsRequest{})
	if err != nil {
		return "", err
	}

	return res.Params.BondDenom, nil
}
```

(basiscally the proto.Marshal, queryFn() and proto.UnMarshal steps are done inside NewQueryServerTestHelper.Invoke already, so we could re-use that)

Reply:
@gsora ^ this is just a suggestion, I'll let you make the decision if to take it or not. I already approved the PR, and your method with hardcoded paths looks fine to me too 👍

Reply:
Looks cleaner for sure, but I don't feel great using something made for unit tests in non-test code.

Reply:
App should expose the `QueryServer`. So I fully support to make a related update (maybe in other PR)? Not sure if this should block this PR, if we don't want to implement it here, then let's create a task to:
- properly expose QueryServer
- update this code